### PR TITLE
Add Integration tests

### LINKED ISSUE #7: Integration tests

### PR DESCRIPTION
**Pull request by ArtMironow**

### CODE REVIEW

File: Meetme.ProfileService/Meetme.ProfileService.API/appsettings.json [hunk 1]
```diff
@@ -6,7 +6,8 @@
         }
     },
     "ConnectionStrings": {
-        "DefaultConnection": "Host=localhost;Port=5432;Database=profileServiceDb;Username=postgres;Password=A12345b$"
+        "DefaultConnection": "Host=localhost;Port=5432;Database=profileServiceDb;Username=postgres;Password=A12345b$",
+        "TestConnection": "Host=localhost;Port=5432;Database=profileServiceTestDb;Username=postgres;Password=A12345b$"
```
Comment (moltenessence) [inline Meetme.ProfileService/Meetme.ProfileService.API/appsettings.json:10]: you could use just in-memory db

File: Meetme.ProfileService/Meetme.ProfileService.BLL/Services/ProfileService.cs [hunk 2]
```diff
@@ -73,7 +73,7 @@ public async Task UpdateAsync(Guid id, UpdateProfileModel model, CancellationTok
 
 		if (profile == null)
 		{
-			throw new BusinessLogicException("Profile does not exist");
+			throw new KeyNotFoundException("Profile does not exist");
```
Comment (moltenessence) [inline Meetme.ProfileService/Meetme.ProfileService.BLL/Services/ProfileService.cs]: what is "KeyNotFoundException" ?

File: Meetme.ProfileService/Meetme.ProfileService.IntegrationTests/ProfilesControllerTests.cs [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+public class ProfilesControllerTests : BaseIntegrationTest
+{
+	public ProfilesControllerTests(MeetmeProfileServiceWebApplicationFactory factory) : base(factory)
+	{
+	}
+
+	[Theory, OmitOnRecursionAutoData]
+	public async Task CreateAsync_CreatesProfile(CreateProfileViewModel viewModel)
+	{
+		var response = await httpClient.PostAsJsonAsync(EndPointKeys.Profiles, viewModel, default);
```
Comment (moltenessence) [inline Meetme.ProfileService/Meetme.ProfileService.IntegrationTests/ProfilesControllerTests.cs]: test methods names do not reflect condition: why or when it creates profile, when it returns bad request and so on..

File: Meetme.ProfileService/Meetme.ProfileService.IntegrationTests/ProfilesControllerTests.cs [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+		var responseContent = await response.Content.ReadFromJsonAsync<IEnumerable<ProfileViewModel>>();
+		responseContent?.Count().ShouldBe(profileCount);
+	}
+
+	[Theory, OmitOnRecursionAutoData]
+	public async Task GetByIdAsync_ReturnsProfile(ProfileEntity profileEntity)
+	{
+		await dbContext.Set<ProfileEntity>().AddAsync(profileEntity, default);
+		await dbContext.SaveChangesAsync();
+
```
Comment (moltenessence) [inline Meetme.ProfileService/Meetme.ProfileService.IntegrationTests/ProfilesControllerTests.cs]: Move such logic as Db seeding to separate methods

**Review by moltenessence — CHANGES_REQUESTED** [review]

File: Meetme.ProfileService/Meetme.ProfileService.IntegrationTests/ProfilesControllerTests.cs [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+	[Theory, OmitOnRecursionAutoData]
+	public async Task GetAllAsync_ReturnsProfiles_WhenProfilesExist(IEnumerable<ProfileEntity> profileEntities)
+	{
+		var profileCount = profileEntities.Count();
+
+		foreach (var profileEntity in profileEntities)
+		{
+			await PopulateProfile(profileEntity);
+		}
+
```
Comment (moltenessence) [inline Meetme.ProfileService/Meetme.ProfileService.IntegrationTests/ProfilesControllerTests.cs]: AddRange?

File: Meetme.ProfileService/Meetme.ProfileService.IntegrationTests/ProfilesControllerTests.cs [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+		response.StatusCode.ShouldBe(HttpStatusCode.OK);
+
+		var responseContent = await response.Content.ReadFromJsonAsync<IEnumerable<ProfileViewModel>>();
+		responseContent?.Count().ShouldBe(profileCount);
+	}
+
+	[Theory, OmitOnRecursionAutoData]
+	public async Task GetByIdAsync_ReturnsProfile_WhenProfileExists(ProfileEntity profileEntity)
+	{
+		await PopulateProfile(profileEntity);
```
Comment (moltenessence) [inline Meetme.ProfileService/Meetme.ProfileService.IntegrationTests/ProfilesControllerTests.cs]: AddProfile
InitializeProfileData

File: Meetme.ProfileService/Meetme.ProfileService.IntegrationTests/ProfilesControllerTests.cs [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+	[Theory, OmitOnRecursionAutoData]
+	public async Task UpdateAsync_UpdatesProfile_WhenProfileExists(UpdateProfileViewModel updateProfileViewModel, ProfileEntity profileEntity)
+	{
+		await PopulateProfile(profileEntity);
+
+		var response = await httpClient.PutAsJsonAsync(EndPointKeys.Profiles + $"/{profileEntity.Id}", updateProfileViewModel, default);
+
+		response.StatusCode.ShouldBe(HttpStatusCode.OK);
+	}
+
```
Comment (moltenessence) [inline Meetme.ProfileService/Meetme.ProfileService.IntegrationTests/ProfilesControllerTests.cs]: where is check that data was really updated?

File: Meetme.ProfileService/Meetme.ProfileService.IntegrationTests/ProfilesControllerTests.cs [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+	[Theory, OmitOnRecursionAutoData]
+	public async Task DeleteAsync_DeletesProfile_WhenProfileExists(ProfileEntity profileEntity)
+	{
+		await PopulateProfile(profileEntity);
+
+		var response = await httpClient.DeleteAsync(EndPointKeys.Profiles + $"/{profileEntity.Id}");
+
+		response.StatusCode.ShouldBe(HttpStatusCode.OK);
+	}
+
```
Comment (moltenessence) [inline Meetme.ProfileService/Meetme.ProfileService.IntegrationTests/ProfilesControllerTests.cs]: where are tests for /POST?

**Review by moltenessence — CHANGES_REQUESTED** [review]

**Review by daniilbluetape — APPROVED** [review]

**Pull request merged**